public/login.js

```diff
@@ -31,23 +50,33 @@ function login(event) {
   const urlParams = new URLSearchParams(window.location.search)
   const identityProviderUrl =
     urlParams.get('identity_provider_url') || 'https://api.core.ac.uk/'
+  const fallbackIDP =
+    urlParams.get('fallback_idp') || 'https://dashboard.core.ac.uk/'
 
   window.dispatchEvent(new Event('login-processing'))
   const formData = new FormData(event.target)
   const data = new URLSearchParams(formData)
-  const url = new URL('login_check', identityProviderUrl)
-  fetch(url, {
-    method: 'POST',
-    body: data,
-    credentials: 'include',
-  })
-    .then(r => {
-      if (r.status === 401) showWrongCredentialsMessage()
-    })
-    .finally(() => {
+  const loginUrl = new URL('login_check', identityProviderUrl)
+  performLoginRequest(loginUrl, data)
+    .then(
+      () => 'login-processing',
+      error => {
+        if (error instanceof ForbiddenError) {
+          const fallbackUrl = new URL('login_check', fallbackIDP)
+          return performLoginRequest(fallbackUrl, data).then(
```

Comment:
```suggestion
          return performLoginRequest(fallbackUrl, data).finally(() => 'login-fallback')
```

Reply:
What is the difference? I am not sure I can chain `then` to `finaly` and get the result from `finaly`'s callback.

Reply:
Never mind I misread the code. Sorry about that